Fix/delete courses from study plan

Carlgm-18 opened the pull request.

Carlgm-18 committed:
fix: delete plan uses que correct request type

Carlgm-18 committed:
fix: minor style fix

Carlgm-18 committed:
fix: improved the way of editing an study plan courses

Carlgm-18 committed:
fix: linter errors

Carlgm-18 committed:
fix: change type for linter help

Carlgm-18 committed:
feat: see which courses need prerequisites

Carlgm-18 committed:
feat: added study-plan validation request

acata8 reviewed (changes requested):
Hi Carlos, I checked the functionality that you edited.
I tried to remove some courses from a study plan that i created and the removing works correctly, but when if I save the plan after a course deletion the study plan doesn't store any course anymore, even if I have some selected courses inside.
I don't know if the functionality of saving was already broken, but maybe it's better to fix it because it's strictly related.

Carlgm-18 committed:
fix: correctly add and delete courses to study plan

Carlgm-18 committed:
Merge branch 'dev' into fix/delte-courses-from-study-plan

Carlgm-18 commented:
I have fixed the problem.

Carlgm-18 committed:
fix: format inconsistencies

feat: added a new type to work with the study plan validation and implemented the function to fetch a course by id

Carlgm-18 committed:
fix: exported the modal renderization into its own function

Carlgm-18 committed:
feat: map the not_satisfied_prerequisites from an string array to a Course array

Carlgm-18 committed:
Merge branch 'dev' into feat/study-plan-requirements-validation

Carlgm-18 committed:
feat: added the semester property to the Course interface

Carlgm-18 committed:
feat: added semester info

Carlgm-18 committed:
fix: refactor to only use one Course property

Carlgm-18 committed:
fix: refactor to only use one Course property

Carlgm-18 committed:
fix: using the Course interface from index.tsx

Carlgm-18 committed:
Merge branch 'dev' into feat/include-semester-data

Carlgm-18 committed:
Merge branch 'feat/include-semester-data' into fix/delte-courses-from-study-plan
